articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+Bulk Passwords Checking
+=======================
+
+In Auth0 we focus on bring you the tools you need to keep your systems and your users safe.
+Bulk passwords checking is a tool that allows you to check your user's passwords against pre-defined list of
+passwords to test their strength.
+
+This tool together with our [Breached Password Security](/anomaly-detection/breached-passwords) helps you audit your
+users security and take actions to protect them.
+
+## Bulk Passwords Checking vs Breached Password Security
+Breached Password Security tools checks user passwords against our database of compromised passwords, those passwords usually
+come from security breached experienced by a third party application that have been released and so they might have been accessed by attackers. If a compromised account is detected different actions may be taken based on your configuration.
+
+On the other hand, bulk passwords checking aims to give you the power to audit your users security against your own
+passwords list, those passwords may come from different sources, examples of possible sources might be list of passwords
+available on the Internet (also known as passwords-dictionaries), your own hand-made checking list, etc.
+
+Combined with other tools we offer, actively auditing your users' account passwords will help you to proactively
+protect them and take risk mitigation actions in case a weak or compromised password gets detected.
+
+## Security considerations
+This tool helps you audit and protect your users against weak or compromised passwords, however, if used inappropriately, it might allow an attacker to brute force your users; because of this is really important for you to keep your credentials secure and not to issue tokens that allows to execute bulk passwords checking if not strictly necessary, also, make sure to include a token identifier and set a short expiration period on the tokens that allow this action, that way you will be able to blacklist them if they get compromised. Finally, delete the data once you have finished; that way you can be sure nobody else will be able to access it.
+
+We also take explicit measures to protect your data:
+- As everything in Auth0 all the data is transmitted over TLS even in our internal systems.
+- Your data will be in our systems as less time as possible, only the time needed to
```

Comment:
The data is held only for the time it takes to process and return the results, and then it will be automatically deleted after the results are retrieved. Even so,
 it is considered good practice to manually delete the data once you have retrieved the results (see below).